Feature/mouse aiming

killerwolf162 opened the pull request.

killerwolf162 committed:
Added:
- Function to set MainCamera to PlayerPosition
- Functions to get Bullet rotation and direction when shooting

killerwolf162 committed:
Changed:
- Code cleanup

killerwolf162 committed:
Changed:
- Naming to conform to convention

killerwolf162 committed:
Changed:
- Removed Debug,Log lines

killerwolf162 committed:
Changed:
- Removed unnecesairy comment

killerwolf162 committed:
Changed:
- fixed typo

GodlyHamster commented on Assets/Scripts/Bullet/Bullet.cs:
```diff
@@ -57,27 +64,42 @@ public void Decorate(BulletDecorator decorator)
         GameHandler.instance.Subscribe(this);
-        _startPosition = bullet.transform.position;
+
+        bullet.transform.position = _mainCam.transform.position + new Vector3(0,0,1);
```
Wat moet dit precies doen? al moet dit de bullet achter de speler renderen: graag de rendering layer van bullet/player aanpassen

GodlyHamster commented on Assets/Scripts/Bullet/Bullet.cs:
```diff
@@ -57,27 +64,42 @@ public void Decorate(BulletDecorator decorator)
         timer = 0;
     }
 
+    public Vector3 GetAimDirection()
+    {
+        _mousePos = _mainCam.ScreenToWorldPoint(Input.mousePosition);
+        var directionToGive = _mousePos - bullet.transform.position;
+
+        return directionToGive;
+    }
+
+    public Quaternion GetBulletRotation()
+    {
+        _mousePos = _mainCam.ScreenToWorldPoint(Input.mousePosition);
+        Vector3 rotation = _mousePos - bullet.transform.position;
+        float zRotation = Mathf.Atan2(rotation.y, rotation.x) * Mathf.Rad2Deg;
+
+        return Quaternion.Euler(0, 0, zRotation - 90);
+    }
+
```
Graag de aimdirection van de player opvragen.
Stel andere objecten(enemies) kunnen ook schieten dan schieten ze niet vanuit de camera/player positie naar waar de speler richt

GodlyHamster commented on Assets/Scripts/GameHandler.cs:
```diff
@@ -9,19 +9,21 @@ public class GameHandler : MonoBehaviour
     [SerializeField] private GameObject _playerPrefab;
     [SerializeField] private int _bulletDamage;
-    [SerializeField] private Color baseBulletColor;
+    [SerializeField] private Color _baseBulletColor;
```
Kan bullet color niet beter bijgehouden worden door de bullet zelf?
Dan kunnen de bullet decoraters deze ook makkelijk aanpassen

GodlyHamster commented on Assets/Scripts/GameHandler.cs:
```diff
@@ -9,19 +9,21 @@ public class GameHandler : MonoBehaviour
 
-    private Bullet _bullet;
-    
+    private Bullet _bullet;  
```
Moet bullet in deze scope leven?
Volgens mij kan bullet beter worden verplaatst naar de CreateBullet() functie

killerwolf162 committed:
Added:
- IShooter interface
- Resource folder

Changed:
- Create bullet function is in Playercontroller instead of GameHandler
- PlayerController gives Rotation and Direction to bullet when activated via IShooter, this makes it so Bullet.cs can also be used by enemys if they have this interface
